singleinflight.go

```diff
@@ -49,9 +51,11 @@ func (g *singleflight) Do(key string, fn func() (*Msg, time.Duration, error)) (v
 	c.val, c.rtt, c.err = fn()
 	c.wg.Done()
 
-	g.Lock()
-	delete(g.m, key)
-	g.Unlock()
+	if !g.dontDeleteForTesting {
```

Comment:
this is maybe a bit of a nit, but can we do this without changing the non-test code? Wrap this in a new struct for testing and override this function?

Reply:
I didn’t love that either. I’m not sure I see how though. This seems like the most minimal change to me, but I’m open to any suggestions.

Reply:
I didn’t love that either. I’m not sure I see how though. This seems like the most minimal change to me, but I’m open to any suggestions.